### PR TITLE
Run Coveralls in debug mode and pin dependency version(#2258)

### PR DESCRIPTION
**Pull request by achave11**
Author (check before every review)

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue
- [x] Added `DCP/1` label                                   <sub>or this PR does not target an `hca/*` branch</sub>
- [x] Created issue to track porting these changes          <sub>or this PR does not need porting</sub> 
- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>
- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in chain            <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>
- [x] Ran `make requirements_update`                        <sub>or this PR leaves requirements*.txt, common.mk and Makefile untouched</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR leaves requirements*.txt untouched</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR leaves requirements*.txt untouched</sub>
- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>
- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`

Primary reviewer (before pushing merge commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Build passes in `sandbox`                             <sub>or added `no sandbox` label</sub>
- [x] Reindexed `sandbox`                                   <sub>or this PR does not require reindexing</sub>
- [x] Added PR reference to merge commit
- [x] Moved linked issue to Merged
- [x] Pushed merge commit to Github

Primary reviewer (after pushing merge commit)

- [ ] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened chain                                       <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Commented on demo expectations                        <sub>or labeled as `no demo`</sub>
- [ ] Pushed merge commit to Gitlab                         <sub>or this changes can be pushed later, together with another PR</sub>
- [ ] Reindexed `dev`                                       <sub>or this PR does not require reindexing</sub>
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Unassign reviewer from PR

### CODE REVIEW

**Review by natanlao — APPROVED** [review]
👍

File: .travis.yml [hunk 1]
```diff
@@ -73,6 +73,6 @@ jobs:
       # Currently not idempotent (https://github.com/lemurheavy/coveralls-public/issues/1435)
       script:
         - set -e
-        - pip install coveralls
+        - pip install coveralls==2.1.2
         - ln $COVERAGE_FILE .coverage
-        - coveralls
+        - coveralls --verbose
```
Comment (achave11) [inline .travis.yml:78]: Note:
Using the flag `debug` wouldn't publish the results back to the Github, but would show all check marks in green.
Using `--verbose` had the same logging output effect, with the GH report being honored.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
It looks like this logs the entire source code.

https://travis-ci.com/github/DataBiosphere/azul/jobs/417556620#L618

That's too verbose.

**Comment by achave11** [comment]
> It looks like this logs the entire source code.
> 
> https://travis-ci.com/github/DataBiosphere/azul/jobs/417556620#L618
> 
> That's too verbose.

@hannes-ucsc I forwarded the email with Coveralls.io support. 
As for your concern, agreed. We are sending all of our source code (to be checked for coverage) wrapped in JSON through the Coveralls API which is what we are [seeing](https://travis-ci.com/github/DataBiosphere/azul/jobs/417556620#L618). We also see the response from Coveralls and it might include error messages returned by the API if there is a failure.
I can not conceive a viable solution that excludes our source code but preserves the response from the API. I was unable to find any useful information in the [coveralls-cli source code](https://github.com/coveralls-clients/coveralls-python/blob/2.1.2/coveralls/cli.py) or their [readthedocs documentation](https://coveralls-python.readthedocs.io/en/latest/troubleshooting.html). On the other hand, it is nice that the Travis unit test run is isolated from the Coveralls one, it does not clutter the unit test report which I think we get more value out off on a consistent basis.

**Comment by hannes-ucsc** [comment]
See if grep -v can be used to filter coverall's output.

**Comment by achave11** [comment]
The first line with `source_files` can safely be completely remove; failure output comes at a latter stage. 

```
$ coveralls --verbose 2>&1 | grep -v '^{"source_files":'
Missing .coveralls.yml file. Using only env variables.
Submitting coverage to coveralls.io...
==
Reporting 131 files
==

lambdas/indexer/app.py - 56/136
lambdas/service/app.py - 348/2576
src/azul/__init__.py - 339/1074
src/azul/aws_service_model.py - 0/36
src/azul/azulclient.py - 68/371
src/azul/bigquery.py - 5/12
src/azul/chalice.py - 60/147
src/azul/changelog.py - 32/66
src/azul/collections.py - 20/235
src/azul/deployment.py - 97/315
src/azul/doctests.py - 4/27
src/azul/drs.py - 41/172
src/azul/dss.py - 45/249
src/azul/es.py - 22/64
src/azul/exceptions.py - 8/65
src/azul/files.py - 23/60
src/azul/health.py - 120/260
src/azul/hmac.py - 27/47
src/azul/http.py - 4/9
src/azul/indexer/__init__.py - 24/69
src/azul/indexer/aggregate.py - 149/412
src/azul/indexer/document.py - 252/601
src/azul/indexer/document_service.py - 26/99
src/azul/indexer/index_controller.py - 150/297
src/azul/indexer/index_service.py - 216/658
src/azul/indexer/lambda_iam_policy.py - 0/122
src/azul/indexer/transform.py - 7/70
src/azul/json.py - 14/60
src/azul/json_freeze.py - 38/154
src/azul/lambda_layer.py - 0/89
src/azul/lambdas.py - 10/65
src/azul/logging.py - 21/40
src/azul/modules.py - 17/45
src/azul/objects.py - 12/68
src/azul/openapi/__init__.py - 14/80
src/azul/openapi/params.py - 15/70
src/azul/openapi/responses.py - 8/43
src/azul/openapi/schema.py - 60/434
src/azul/plugins/__init__.py - 49/319
src/azul/plugins/metadata/__init__.py - 0/0
src/azul/plugins/metadata/hca/__init__.py - 13/360
src/azul/plugins/metadata/hca/aggregate.py - 44/185
src/azul/plugins/metadata/hca/full_metadata.py - 109/251
src/azul/plugins/metadata/hca/transform.py - 241/940
src/azul/plugins/repository/__init__.py - 0/0
src/azul/plugins/repository/dss/__init__.py - 88/464
src/azul/plugins/repository/tdr/__init__.py - 139/597
src/azul/portal_service.py - 80/219
src/azul/queues.py - 68/453
src/azul/remote_debug.py - 0/163
src/azul/requests.py - 0/24
src/azul/security/__init__.py - 0/0
src/azul/security/authenticator.py - 114/218
src/azul/service/__init__.py - 22/62
src/azul/service/async_manifest_service.py - 62/146
src/azul/service/buffer.py - 34/67
src/azul/service/cart_export_job_manager.py - 51/114
src/azul/service/cart_export_service.py - 34/82
src/azul/service/cart_item_manager.py - 139/377
src/azul/service/collection_data_access.py - 80/161
src/azul/service/drs_controller.py - 130/398
src/azul/service/dynamo_data_access.py - 82/231
src/azul/service/elasticsearch_service.py - 191/760
src/azul/service/hca_response_v5.py - 97/721
src/azul/service/index_query_service.py - 43/184
src/azul/service/lambda_iam_policy.py - 0/189
src/azul/service/manifest_service.py - 440/999
src/azul/service/repository_controller.py - 57/121
src/azul/service/step_function_helper.py - 13/62
src/azul/service/storage_service.py - 125/269
src/azul/service/user_service.py - 23/50
src/azul/service/utilities.py - 2/15
src/azul/strings.py - 14/48
src/azul/subscription.py - 0/71
src/azul/template.py - 6/32
src/azul/terra.py - 54/209
src/azul/threads.py - 81/253
src/azul/time.py - 29/108
src/azul/types.py - 21/97
src/azul/uuids.py - 25/85
src/azul/vendored/__init__.py - 0/0
src/azul/vendored/frozendict/__init__.py - 25/90
src/azul/version_service.py - 28/78
test/app_test_case.py - 61/131
test/azul_test_case.py - 129/336
test/docker_container_test_case.py - 42/104
test/dynamo_test_case.py - 13/26
test/es_test_case.py - 40/79
test/health_check_test_case.py - 132/294
test/indexer/__init__.py - 75/148
test/indexer/test_hca_indexer.py - 555/1340
test/indexer/test_health_check.py - 22/50
test/indexer/test_indexer_controller.py - 76/161
test/indexer/test_projects.py - 53/106
test/integration_test.py - 0/872
test/retorts.py - 95/223
test/security/__init__.py - 0/0
test/security/test_authenticator.py - 151/216
test/service/__init__.py - 42/94
test/service/test_async_manifest_service.py - 137/290
test/service/test_auth_endpoints.py - 68/103
test/service/test_buffer.py - 55/69
test/service/test_cart_export_endpoints.py - 191/347
test/service/test_cart_export_job_manager.py - 90/316
test/service/test_cart_export_service.py - 87/155
test/service/test_cart_item_manager.py - 241/567
test/service/test_collection_data_access.py - 124/205
test/service/test_drs.py - 125/286
test/service/test_dss_proxy.py - 108/209
test/service/test_dynamo_accessor.py - 91/230
test/service/test_health_check.py - 31/61
test/service/test_manifest.py - 175/1497
test/service/test_pagination.py - 47/249
test/service/test_portal_service.py - 74/180
test/service/test_query_shortener.py - 65/135
test/service/test_repository_projects.py - 24/90
test/service/test_repository_specimen.py - 18/56
test/service/test_request_builder.py - 58/338
test/service/test_request_logging.py - 19/45
test/service/test_request_validation.py - 154/280
test/service/test_response.py - 216/2077
test/service/test_storage_service.py - 96/180
test/service/test_tdr.py - 168/453
test/service/test_user_service.py - 64/125
test/service/test_version_service.py - 24/47
test/test_app_logging.py - 74/151
test/test_azul_flake8.py - 50/204
test/test_changelog.py - 22/35
test/test_doctests.py - 35/89
test/test_openapi.py - 75/233
test/version_table_test_case.py - 15/41
Could not submit coverage: 422 Client Error: Unprocessable Entity for url: https://coveralls.io/api/v1/jobs
Traceback (most recent call last):
  File "/Users/abrahamchavez/PycharmProjects/Azul/workspace/.venvexp/lib/python3.8/site-packages/coveralls/api.py", line 205, in wear
    response.raise_for_status()
  File "/Users/abrahamchavez/PycharmProjects/Azul/workspace/.venvexp/lib/python3.8/site-packages/requests/models.py", line 941, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 422 Client Error: Unprocessable Entity for url: https://coveralls.io/api/v1/jobs

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/abrahamchavez/PycharmProjects/Azul/workspace/.venvexp/lib/python3.8/site-packages/coveralls/cli.py", line 85, in main
    result = coverallz.wear()
  File "/Users/abrahamchavez/PycharmProjects/Azul/workspace/.venvexp/lib/python3.8/site-packages/coveralls/api.py", line 208, in wear
    raise CoverallsException('Could not submit coverage: {}'.format(e))
coveralls.exception.CoverallsException: Could not submit coverage: 422 Client Error: Unprocessable Entity for url: https://coveralls.io/api/v1/jobs
```

**Comment by hannes-ucsc** [comment]
For demo, show Travis job output and run coveralls locally with `--verbose` to show what grep to filters out.

**Pull request merged**